From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <tbashir@brightsign.biz>
Date: Tue, 31 Mar 2026 11:50:49 +0100
Subject: feat: Enable HLS support when Brightsign media player is enabled
 OS-20673

Brightsign media player can play HLS streams, enable mimetype support
to enable Youtube live streams through HLS

diff --git a/media/base/BUILD.gn b/media/base/BUILD.gn
index 4e4188156f34b279c39a3e81986254dea562dc86..66517636e634e96bb9f6d36abc831a9294d1f6e1 100644
--- a/media/base/BUILD.gn
+++ b/media/base/BUILD.gn
@@ -440,6 +440,10 @@ source_set("base") {
     deps += [ "//third_party/libvpx" ]
   }
 
+  if (enable_brightsign_media_player) {
+    defines += [ "ENABLE_BRIGHTSIGN_MEDIA_PLAYER" ]
+  }
+
   if (enable_libaom) {
     sources += [
       "libaom_thread_wrapper.cc",
diff --git a/media/base/mime_util_internal.cc b/media/base/mime_util_internal.cc
index 30ef5e1ac00dbf0faca7a713a1d6e241ea2fef2e..2158c620b97ef9ca79817baf90a7424973a7402f 100644
--- a/media/base/mime_util_internal.cc
+++ b/media/base/mime_util_internal.cc
@@ -407,6 +407,28 @@ void MimeUtil::AddSupportedMediaFormats() {
   video_3gpp_codecs.emplace(H264);
   AddContainerWithCodecs("video/3gpp", video_3gpp_codecs);
 
+#if defined(ENABLE_BRIGHTSIGN_MEDIA_PLAYER)
+  bool is_brightsign_player =
+      base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kUseBrightsignMediaPlayer);
+  if (is_brightsign_player) {
+    CodecSet hls_codecs{H264,
+                        // TODO(ddorwin): Is any MP3 codec string variant
+                        // included in real queries?
+                        MP3,
+                        // Android HLS only supports MPEG4_AAC (missing demuxer
+                        // support for MPEG2_AAC)
+                        MPEG4_AAC};
+    AddContainerWithCodecs("application/x-mpegurl", hls_codecs);
+    AddContainerWithCodecs("application/vnd.apple.mpegurl", hls_codecs);
+    AddContainerWithCodecs("audio/mpegurl", hls_codecs);
+    // Not documented by Apple, but unfortunately used extensively by Apple and
+    // others for both audio-only and audio+video playlists. See
+    // https://crbug.com/675552 for details and examples.
+    AddContainerWithCodecs("audio/x-mpegurl", hls_codecs);
+  }
+#endif
+
 #if BUILDFLAG(IS_ANDROID) || BUILDFLAG(ENABLE_HLS_DEMUXER)
   bool can_play_hls = false;
 #endif
